### PR TITLE
added mclag sonic yang

### PR DESCRIPTION
**Pull request by gitsabari**
#### How I did it
Added mclag sonic yang file for the MCLAG enhancements  as per HLD: Azure/SONiC#596

#### How to verify it
try rest APIs

#### Description for the changelog
Added mclag sonic yang

### CODE REVIEW

**Comment by rlhui** [comment]
Tests are missing?  Thanks.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:44]: Update the revision date.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:66]: Please add descriptions for all fields.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:82]: uint8?

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:122]: what for this field is used?

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:151]: no disable option? this is not clear, please fix it.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:161]: Not sure if we need to add any model for STATE_DB now, please remove it.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:122]: key itself contains MCLAG Domain id and MCLAG Portchannel. if_type is a dummy field to create the instance

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:151]: okay i will add disable option

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:161]: STATE_DB entry are required for show commands

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:82]: currently this is the implementation we have. this is just a just to accomodate a long keepalive interval

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:66]: sure will add

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:44]: sure will do

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:104]: We need the UT cases to cover various conditions present in the YANG.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:53]: Wondering if we really need list with max-elements 1, can you explain any potential use-case that might come up in the future with more than one MCLAG domains?

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:104]: working on adding UT cases. will check in soon

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:53]: this is added for future proof to support multi domain config

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:112]: added some base case, working on adding some more

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:167]: Ok, then I don't see a need for unique_ip field, if the entry is present, unique IP is enabled on certain VLAN, if not present, unique IP is not enabled.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:179]: We have not added any STATE_DB tables in the YANG model so far, this is something we need to discuss and decide in the community meeting, if required, we need to plan for adding other STATE_DB tables as well.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:90]: we can accommodate higher range by simply modifying from uint8 to uint16 in the future, no issues in that, but for a given range now, uint8 is good enough.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:134]: redis DB can have the entry with NONE:NONE when no field is present for an entry, this dummy field is not required.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:179]: thanks for the comments. currently sonic-extension changes for db-name, key-delimit etc is not yet into community. so i will remove state db entries which requires this and add it later when those are available

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:164]: yes you right from that perspective. But since REDIS DB config db table entry cannot be empty. This field is required in that way

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:110]: @venkatmahalingam 
added UT Test cases. thanks

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (lguohan) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:7]: can you check the space to be consistent? 4 spaces

File: src/sonic-yang-models/yang-models/sonic-types.yang
Comment (lguohan) [inline src/sonic-yang-models/yang-models/sonic-types.yang:61]: where is this getting used?

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (lguohan) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:66]: do we need to add validation for the source ip

File: src/sonic-yang-models/tests/files/sample_config_db.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/files/sample_config_db.json:1040]: add this configuration in https://github.com/Azure/SONiC/wiki/Configuration as source of truth, if not added already.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (lguohan) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:74]: add mandatory fields.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (lguohan) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:55]: this means we can only have one mclag domain right now.

**Comment by praveen-li** [comment]
AI: DPB config links @praveen-li

**Comment by praveen-li** [comment]
> AI: DPB config links @praveen-li

https://github.com/Azure/sonic-utilities/blob/679a4ba602cc4c1ea77b52e1c782e8663d99e03b/tests/config_dpb_test.py#L715 [Build time Config for front end DPB test]

File: src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json:11]: An invalid test must have eStr field, I think. Is this test passing right now.

File: src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json:15]: Same for this test

File: src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json:19]: If you have a error tag or message in Yang file, then libyang will throw different error message. In that case, current eStrKey as "must" is not correct.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/mclag.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/mclag.json:2]: Kindly see if this config can be smaller.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/mclag.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/mclag.json:171]: Kindly see if this config also can be reduced.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:24]: "lag" prefix is used in earlier Yang files, can we please stick to that.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:28]: same for this...kindly refer other files and match the prefix value.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:40]: One PR should have only one revision field. We do not need separate field for ""Addressed review comments". Thx

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:78]: Add MCLAG config at https://github.com/Azure/sonic-utilities/blob/679a4ba602cc4c1ea77b52e1c782e8663d99e03b/tests/config_dpb_test.py#L715 
to test impact on DPB. Thx

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:144]: Do we have test for this must condition?

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:150]: Do we have test for this must condition?

**Comment by alexrallen** [comment]
In order to test the functionality of DPB....

1. Setup MCLAG across some ports (i.e. Ethernet0 and Ethernet4 for example sake)
2. Make sure configuration is applied and MCLAG is functioning as expected
3. Run `show interface breakout` to get a full list of supported breakout modes per port
4. Look for the `Ethernet0` key in the results and look for `breakout_modes` under that
5. Pick one of the breakout modes and copy the key
6. Run `config interface breakout Ethernet0 2x50G[25G,10G] -y -f` where the interface is the one you are breaking out and the mode is copied straight from step 5
7. The command should complete without error
8. Run `show interface status` and verify the port has been successfully broken out
9. Run `config save -y`
10. Verify there are no references to the original port in the config (particularly MCLAG config sections)

If this process works without a problem then you can be confident that DPB should work. One thing also to consider, as is mentioned in the yang group meetings is that you should include all possible config options for MCLAG to ensure that Yang is not hung up on anything it may not recognize. 

Please let me know if there are any questions.

**Comment by zhangyanzhao** [comment]
BRCM team will handle this one.

**Comment by zhangyanzhao** [comment]
BRCM team is working on this and ETA is 9/30

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:7]: done

File: src/sonic-yang-models/yang-models/sonic-types.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-types.yang]: removed. not used

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:62]: done

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:51]: yes only one domain is allowed now

File: src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json
Comment (gitsabari) [reply src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json:12]: done

File: src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json
Comment (gitsabari) [reply src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json]: done

File: src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json
Comment (gitsabari) [reply src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json]: done

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang]: done

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang]: done

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:140]: yes we do

**Comment by gitsabari** [comment]
> In order to test the functionality of DPB....
> 
> 1. Setup MCLAG across some ports (i.e. Ethernet0 and Ethernet4 for example sake)
> 2. Make sure configuration is applied and MCLAG is functioning as expected
> 3. Run `show interface breakout` to get a full list of supported breakout modes per port
> 4. Look for the `Ethernet0` key in the results and look for `breakout_modes` under that
> 5. Pick one of the breakout modes and copy the key
> 6. Run `config interface breakout Ethernet0 2x50G[25G,10G] -y -f` where the interface is the one you are breaking out and the mode is copied straight from step 5
> 7. The command should complete without error
> 8. Run `show interface status` and verify the port has been successfully broken out
> 9. Run `config save -y`
> 10. Verify there are no references to the original port in the config (particularly MCLAG config sections)
> 
> If this process works without a problem then you can be confident that DPB should work. One thing also to consider, as is mentioned in the yang group meetings is that you should include all possible config options for MCLAG to ensure that Yang is not hung up on anything it may not recognize.
> 
> Please let me know if there are any questions.

@

> In order to test the functionality of DPB....
> 
> 1. Setup MCLAG across some ports (i.e. Ethernet0 and Ethernet4 for example sake)
> 2. Make sure configuration is applied and MCLAG is functioning as expected
> 3. Run `show interface breakout` to get a full list of supported breakout modes per port
> 4. Look for the `Ethernet0` key in the results and look for `breakout_modes` under that
> 5. Pick one of the breakout modes and copy the key
> 6. Run `config interface breakout Ethernet0 2x50G[25G,10G] -y -f` where the interface is the one you are breaking out and the mode is copied straight from step 5
> 7. The command should complete without error
> 8. Run `show interface status` and verify the port has been successfully broken out
> 9. Run `config save -y`
> 10. Verify there are no references to the original port in the config (particularly MCLAG config sections)
> 
> If this process works without a problem then you can be confident that DPB should work. One thing also to consider, as is mentioned in the yang group meetings is that you should include all possible config options for MCLAG to ensure that Yang is not hung up on anything it may not recognize.
> 
> Please let me know if there are any questions.

@alexrallen 
thank you for detailed steps. sorry wasnt able to work on it yet. i am planning to get it done this week. mclag doesn't  have dpb related config, except it has mclag portchannel and peer link configuration. do i have to write python script for this and do you have sample/example script which does it. please help with that

**Comment by gitsabari** [comment]
@praveen-li, @arlakshm, @venkatmahalingam
seems to be a generic build failure

2021-11-10T09:16:34.4789523Z     def readIjsonInput(self, yang_test_file, test):
2021-11-10T09:16:34.4790063Z         try:
2021-11-10T09:16:34.4790601Z             # load test specific Dictionary, using Key = func
2021-11-10T09:16:34.4791191Z             # this is to avoid loading very large JSON in memory
2021-11-10T09:16:34.4791801Z             print(" Read JSON Section: " + test)
2021-11-10T09:16:34.4792317Z             jInput = ""
2021-11-10T09:16:34.4793116Z             with open(yang_test_file, 'rb') as f:
2021-11-10T09:16:34.4793711Z                 jInst = ijson_itmes(f, test)
2021-11-10T09:16:34.4794240Z >               for it in jInst:
2021-11-10T09:16:34.4794992Z E               ijson.common.IncompleteJSONError: parse error: invalid object key (must be a string)
2021-11-10T09:16:34.4795879Z E                         "up"             },          },         "DEVICE_NEIGHBOR": {
2021-11-10T09:16:34.4796874Z E                                    (right here) ------^
2021-11-10T09:16:34.4797777Z 
2021-11-10T09:16:34.4798653Z tests/libyang-python-tests/test_sonic_yang.py:54: IncompleteJSONError
2021-11-10T09:16:34.4799655Z ----------------------------- Captured stdout call -----------------------------
2021-11-10T09:16:34.4800277Z  Read JSON Section: SAMPLE_CONFIG_DB_UNKNOWN
2021-11-10T09:16:34.4801048Z Reading Ijson failed
2021-11-10T09:16:34.4801599Z =============================== warnings summary ===============================
2021-11-10T09:16:34.4802325Z sonic_yang_ext.py:319
2021-11-10T09:16:34.4803382Z   /sonic/src/sonic-yang-mgmt/sonic_yang_ext.py:319: DeprecationWarning: invalid escape sequence \C
2021-11-10T09:16:34.4804203Z     '''
2021-11-10T09:16:34.4804403Z 
2021-11-10T09:16:34.4804780Z sonic_yang_ext.py:357
2021-11-10T09:16:34.4805831Z   /sonic/src/sonic-yang-mgmt/sonic_yang_ext.py:357: DeprecationWarning: invalid escape sequence \C
2021-11-10T09:16:34.4806587Z     '''
2021-11-10T09:16:34.4806797Z 
2021-11-10T09:16:34.4807521Z -- Docs: https://docs.pytest.org/en/latest/warnings.html
2021-11-10T09:16:34.4808224Z =============== 3 failed, 23 passed, 2 warnings in 0.36 seconds ================
2021-11-10T09:16:34.4809264Z [  FAIL LOG END  ] [ target/python-wheels/sonic_yang_mgmt-1.0-py3-none-any.whl ]
2021-11-10T09:16:34.4810478Z make: *** [slave.mk:635: target/python-wheels/sonic_yang_mgmt-1.0-py3-none-any.whl] Error 1
2021-11-10T09:16:34.9718636Z DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
2021-11-10T09:16:38.3216870Z make[1]: *** [Makefile.work:300: target/docker-sonic-vs.gz] Error 2
2021-11-10T09:16:38.3218062Z make[1]: Leaving directory '/agent/_work/2/s'
2021-11-10T09:16:38.3219755Z make: *** [Makefile:32: target/docker-sonic-vs.gz] Error 2
2021-11-10T09:16:38.3297181Z ##[error]Bash exited with code '2'.
2021-11-10T09:16:38.3341401Z ##[section]Finishing: Build sonic image

**Comment by zhangyanzhao** [comment]
build failure looks general, not specific to this PR. Help is wanted.

**Comment by zhangyanzhao** [comment]
conflict need be resolved by @gitsabari

**Comment by gitsabari** [comment]
all merge conflicts and comments addressed. build failure looks general, not specific to this PR. Need help to fix
once fixed this PR can be approved ane merged

**Comment by gitsabari** [comment]
@arlakshm, @zhangyanzhao, @praveen-li , @lguohan 
all review comments addressed. test cases passed. please help with approve and. merging this PR

**Comment by zhangyanzhao** [comment]
@qiluo-msft @arlakshm would you please help to approve and merge? Thanks.

File: src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json
Comment (qiluo-msft) [inline src/sonic-yang-models/tests/yang_model_tests/tests/mclag.json:17]: >	 [](http://example.com/codeflow?start=0&length=1)

Mixing tabs and spaces in this file. Also please reformat the indentation.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:54]: >   [](http://example.com/codeflow?start=31&length=2)

Use one blank instead of two, if you don't need align it with other column.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-mclag.yang:166]: Remove the empty line.

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang]: addressed this comment

File: src/sonic-yang-models/yang-models/sonic-mclag.yang
Comment (gitsabari) [reply src/sonic-yang-models/yang-models/sonic-mclag.yang:165]: addressed this comment

**Comment by gitsabari** [comment]
addressed all review comments. once build is done, help approve and merge

**Comment by gitsabari** [comment]
@zhangyanzhao , there is a generic  build issue. there is a test failure in other modules, not related to this check.
please help get this resolved

2021-12-16T18:15:59.7969236Z   /usr/local/lib/python3.7/dist-packages/pyangbind/lib/yangtypes.py:375: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
2021-12-16T18:15:59.7969931Z     class TypedList(collections.MutableSequence):
2021-12-16T18:15:59.7970140Z 
2021-12-16T18:15:59.7970655Z -- Docs: https://docs.pytest.org/en/latest/warnings.html
2021-12-16T18:15:59.7971102Z ======== 4 failed, 1586 passed, 3 skipped, 8 warnings in 868.26 seconds ========

**Review by qiluo-msft — DISMISSED** [review]
LGTM. Please also check with other reviewers.

**Comment by gitsabari** [comment]
@zhangyanzhao 
unrelated build issue. can you please get some help for this build issue. other than this. this PR can be merged once build done
gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
make: *** [slave.mk:796: target/docker-gbsyncd-vs.gz] Error 2

**Comment by zhangyanzhao** [comment]
Rajesh Vel in BRCM will come back later.

**Comment by gitsabari** [comment]
@zhangyanzhao 
there seems to be generic issue. there are some test cases failing and it is not related to this change
please get some help to resolve this and also this PR can be merged it is already approved by yang. 


======= 4 failed, 1631 passed, 3 skipped, 8 warnings in 1270.60 seconds ========

    def test_extend__path_with_refs_to_children__extended_moves(self):
        self.verify(OperationType.REMOVE,
                    ["PORT"],
                    tc_ops=[{'op':'remove', 'path':'/PORT/Ethernet0'}],
                    ex_ops=[{'op': 'remove', 'path': '/VLAN_MEMBER/Vlan1000|Ethernet0'},
                            {'op': 'remove', 'path': '/ACL_TABLE/NO-NSW-PACL-V4/ports/0'},
                            {'op': 'remove', 'path': '/VLAN_MEMBER/Vlan1000|Ethernet4'},
                            {'op': 'remove', 'path': '/ACL_TABLE/DATAACL/ports/0'},
                            {'op': 'remove', 'path': '/VLAN_MEMBER/Vlan1000|Ethernet8'},
                            {'op': 'remove', 'path': '/ACL_TABLE/EVERFLOWV6/ports/0'},
                            {'op': 'remove', 'path': '/ACL_TABLE/EVERFLOW/ports/0'},
>                           {'op': 'remove', 'path': '/ACL_TABLE/EVERFLOWV6/ports/1'}])

tests/generic_config_updater/patch_sorter_test.py:1690: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/generic_config_updater/patch_sorter_test.py:1708: in verify
    self.verify_moves(ex_ops, moves)
tests/generic_config_updater/patch_sorter_test.py:1712: in verify_moves
    self.assertCountEqual(ex_ops, moves_ops)
E   AssertionError: Element counts were not equal:
E   First has 1, Second has 0:  {'op': 'remove', 'path': '/VLAN_MEMBER/Vlan1000|Ethernet0'}
E   First has 1, Second has 0:  {'op': 'remove', 'path': '/VLAN_MEMBER/Vlan1000|Ethernet4'}
E   First has 1, Second has 0:  {'op': 'remove', 'path': '/VLAN_MEMBER/Vlan1000|Ethernet8'}
------- generated xml file: /sonic/src/sonic-utilities/test-results.xml --------

**Comment by zhangyanzhao** [comment]
LeafRef has a known issue which may be the root cause for the build failure. Try union leaf.

**Comment by ghooo** [comment]
One of the imports in the `.yang` file might be breaking backlinks and hence breaking sonic-utilities UTs. It is a known problem, check issue #9312

**Comment by Praveen-Brcm** [comment]
Hi All the specific sonic yang files covers all the MCLAG configuration commands.

**Review by Praveen-Brcm — APPROVED** [review]

**Review by Praveen-Brcm — DISMISSED** [review]

**Review by Praveen-Brcm — APPROVED** [review]

**Review by arlakshm — APPROVED** [review]
lgtm

**Review by qiluo-msft — APPROVED** [review]

**Comment by judyjoseph** [comment]
@gitsabari Please raise a separate PR for 202111 branch as there are merge conflicts.

**Pull request merged**
